src/wrappers/PhylorefWrapper.js

```diff
@@ -643,49 +660,91 @@ class PhylorefWrapper {
+    // because they shared an equivalentClass expression. To prevent this from
+    // happening, we now set up each phyloreference as a subclass of every logical
+    // expression that evaluates to it. We use the createAdditionalClass() method
+    // to do that, passing it `false` to ensure that it doesn't just reuse existing
+    // additional classes when doing this.
+
     // We might need to make additional JSON-LD.
     // So we reset our additional class counts and records.
     PhylorefWrapper.additionalClassCount = 0;
     PhylorefWrapper.additionalClassesByLabel = {};
     phylorefAsJSONLD.hasAdditionalClass = [];
 
+    // The list of logical expressions generated for this phyloref.
+    let logicalExpressions;
+
     if (internalSpecifiers.length === 0) {
       // We can't handle phyloreferences without at least one internal specifier.
       phylorefAsJSONLD.malformedPhyloreference = 'No internal specifiers provided';
-    } else {
-      // Step 1. Construct an expression for all internal specifiers.
-      let expressionsForInternals;
+    } else if (externalSpecifiers.length > 0) {
+      // If the phyloreference has at least one external specifier, we
+      // can provide a simplified expression for the internal specifier,
+      // in the form:
+      //  phyloref:includes_TU some [internal1] and
+      //  phyloref:includes_TU some [internal2] and ...
+      // To which we can then add the external specifiers.
       if (internalSpecifiers.length === 1) {
-        expressionsForInternals = [
+        logicalExpressions = PhylorefWrapper.createClassExpressionsForExternals(
+          phylorefAsJSONLD,
           PhylorefWrapper.getIncludesRestrictionForTU(internalSpecifiers[0]),
-        ];
+          externalSpecifiers,
+          []
+        );
       } else {
-        expressionsForInternals = PhylorefWrapper.createClassExpressionsForInternals(
-          phylorefAsJSONLD, internalSpecifiers, []
+        const expressionForInternals = {
+          '@type': 'owl:Class',
+          intersectionOf: internalSpecifiers.map(PhylorefWrapper.getIncludesRestrictionForTU),
+        };
+
+        logicalExpressions = PhylorefWrapper.createClassExpressionsForExternals(
+          phylorefAsJSONLD, expressionForInternals, externalSpecifiers, []
         );
       }
+    } else {
+      // We only have internal specifiers. We therefore need to use the algorithm in
+      // PhylorefWrapper.createClassExpressionsForInternals() to create this expression.
+      logicalExpressions = PhylorefWrapper.createClassExpressionsForInternals(
+        phylorefAsJSONLD, internalSpecifiers, []
+      );
+    }
 
-      if (externalSpecifiers.length === 0) {
-        // If we don't have external specifiers, we can just use the expression
-        // for the internal specifier.
-        phylorefAsJSONLD.equivalentClass = expressionsForInternals;
+    if (logicalExpressions && logicalExpressions.length > 0) {
```

Comment:
Is there an advantage of distinguishing in your code between null and an empty list for `logicalExpression`? If not, why not simply use a zero-length `logicalExpressions` array to say "no logical expressions", which would simplify this (and possibly other) condition and protect against future programming errors?

Reply:
Good point! Fixed in 59354ab and b879547ae, with improved comments as well.